src/riak_core_info_service.erl

```diff
@@ -110,7 +110,27 @@
                            {error, term()}.
 
 start_service(Registration, Shutdown, Provider, Handler) ->
-    riak_core_info_service_sup:start_service(Registration, Shutdown, Provider, Handler).
+    case verify_callable([Registration, Shutdown, Handler]) of
+        true ->
+            riak_core_info_service_sup:start_service(Registration, Shutdown, Provider, Handler);
+        {false, {_Module, _Function, _Arity}=Fun} ->
+            {error, {not_callable, Fun}}
+    end.
+
+verify_callable([]) ->
```

Comment:
Channeling Nick to point out that our style guidelines prefer not doing our own recursion.
This could be neatly handled with `lists:all/2` or, for reporting, `lists:filtermap/2`